lib/matplotlib/__init__.py

```diff
@@ -225,7 +225,7 @@ def _is_writable_dir(p):
             t.write(b'1')
         finally:
             t.close()
-    except OSError:
+    except:
```

Comment:
Looks like we are clear on that. Just tried on py2.6.

```
[centos6] [broot@rd22 gust_checks]$ /usr/bin/python
Python 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> raise "SomeString"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: exceptions must be old-style classes or derived from
BaseException, not str
>>>
```

On Mon, Mar 2, 2015 at 4:43 PM, Thomas A Caswell <notifications@github.com>
wrote:

> In lib/matplotlib/__init__.py
> <https://github.com/matplotlib/matplotlib/pull/4165#discussion_r25641395>:
>
> > @@ -225,7 +225,7 @@ def _is_writable_dir(p):
> >              t.write(b'1')
> >          finally:
> >              t.close()
> > -    except OSError:
> > +    except:
>
> I thought the restriction that you could only raise Exception objects was
> new-ish and we support back to 2.6 (but I couldn't figure out quickly when
> that got changed).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4165/files#r25641395>.
>